opus/import/populate_obs_instrument_VGRSS_prof.py

```diff
@@ -0,0 +1,555 @@
+populate_obs_ring_geometry_VGRSS_center_north_based_incidence2_PROF = \
+    populate_obs_ring_geometry_VGRSS_north_based_incidence2_PROF
+populate_obs_ring_geometry_VGRSS_center_north_based_emission1_PROF = \
+    populate_obs_ring_geometry_VGRSS_north_based_emission1_PROF
+populate_obs_ring_geometry_VGRSS_center_north_based_emission2_PROF = \
+    populate_obs_ring_geometry_VGRSS_north_based_emission2_PROF
+
+# Opening angle to observer: the angle between the ring surface to the direction
+# where outgoing photons to the observer. Positive if observer is at the north
+# side of the ring , negative if it's at the south side. In this case, observer
+# is at the north side, so it's ea - 90 (or 90 - inc). For reference, if
+# observer is at the south side, then oa is 90 - ea.
+# Observer is at north, so it's ea - 90 (positive 0-90)
+def populate_obs_ring_geometry_VGRSS_observer_ring_opening_angle1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+
+    return min_ea - 90.
+
+def populate_obs_ring_geometry_VGRSS_observer_ring_opening_angle2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+    max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+
+    return max_ea - 90.
+
+# Ring elevation to observer, same to opening angle except, it's positive if
+# observer is at north side of Jupiter, Saturn, and Neptune, and south side of
+# Uranus. Negative if observer is at south side of Jupiter, Saturn, and Neptune,
+# and north side of Uranus.
+def populate_obs_ring_geometry_VGRSS_observer_ring_elevation1_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+
+    target_name = populate_target_name_from_index(**kwargs)
+    if target_name =='U RINGS':
+        max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+        el = - (max_ea - 90.) # negative
+    else:
+        min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+        el = min_ea - 90. # positive
+
+    return el
+
+def populate_obs_ring_geometry_VGRSS_observer_ring_elevation2_PROF(**kwargs):
+    metadata = kwargs['metadata']
+    index_row = metadata['index_row']
+
+    target_name = populate_target_name_from_index(**kwargs)
+    if target_name =='U RINGS':
+        min_ea = index_row['MINIMUM_EMISSION_ANGLE']
+        el = - (min_ea - 90.) # negative
+    else:
+        max_ea = index_row['MAXIMUM_EMISSION_ANGLE']
+        el = max_ea - 90. # positive
+
+    return el
+
+# Opening angle to solar: the angle between the ring surface to the direction
+# where incoming photons from the source. Positive if source is at the north
+# side of the ring, negative if it's at the south side. In this case, source
+# is at the south side, so it's - (90 - inc). For reference, if source is at the
+# north side, then oa is 90 - inc.
+# Source is at south of the ring, so it's neagtive which is inc - 90
```

Comment:
Isn't the observer at the south?

Reply:
Based on slack discussion, the observer is at north.